### PR TITLE
[#164630993] Enable FLAG_SECURE when debugmode is disabled

### PR DESCRIPTION
**Pull request by fpersico**
FlagSecure

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Could not get info from npm on <a href='https://www.npmjs.com/package/react-native-flag-secure-android'><code>react-native-flag-secure-android</code></a></a>
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#164630993](https://www.pivotaltracker.com/story/show/164630993): Disable background screenshot with sensitive information


New dependencies added: <a href='https://www.npmjs.com/package/react-native-flag-secure-android'><code>react-native-flag-secure-android</code></a>.
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/RootContainer.tsx [hunk 1]
```diff
@@ -64,6 +65,11 @@ class RootContainer extends React.PureComponent<Props> {
   }
 
   public componentDidMount() {
+    if (!this.props.isDebugModeEnabled) {
```
Comment (cloudify) [inline ts/RootContainer.tsx]: shouldn't this be called only on Android?

**Comment by cloudify** [comment]
@fpersico I was wondering, would it make sense to decouple that logic from the `RootComponent` by creating a separated `<FlagSecure />` component?

**Comment by fpersico** [comment]
@cloudify yes it make sense.

File: ts/components/FlagSecure.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import FlagSecure from "react-native-flag-secure-android";
+
+type Props = {
+  isFlagSecureEnabled: boolean;
+};
+
+class FlagSecureComponent extends React.PureComponent<Props> {
```
Comment (cloudify) [inline ts/components/FlagSecure.tsx]: little comment here? :)

File: ts/RootContainer.tsx [hunk 3]
```diff
@@ -117,6 +118,9 @@ class RootContainer extends React.PureComponent<Props> {
     return (
       <Root>
         <StatusBar barStyle="dark-content" />
+        <FlagSecureComponent
```
Comment (cloudify) [inline ts/RootContainer.tsx]: shouldn't this enabled only for Android?

File: ts/components/FlagSecure.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import FlagSecure from "react-native-flag-secure-android";
+
+type Props = {
+  isFlagSecureEnabled: boolean;
+};
+
+/**
+ * Used to enable/disable FLAG_SECURE at runtime.
```
Comment (cloudify) [inline ts/components/FlagSecure.tsx]:
```suggestion
 * On Android, it enables or disables FLAG_SECURE based on the isFlagSecureEnabled prop.
 * On unmount, FLAG_SECURE gets disabled if it was enabled.
```

**Pull request merged**